services/order.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+from typing import Any
+
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from django.db.models import QuerySet
+from django.contrib.auth.password_validation import validate_password
+
+from db.models import Order, Ticket, User, MovieSession
+
+
+def validate_user_input(username: str, password: str) -> None:
+    if not username or len(username) < 3:
+        raise ValidationError("Username must be at least 3 characters long.")
+    validate_password(password)
+
+
+def create_order(
+        tickets: list[dict[str, Any]],
+        username: str,
+        date: str = None
+) -> Order:
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        raise ValidationError(
+            f"User with username '{username}' does not exist."
+        )
+
+    with transaction.atomic():
+        order = Order.objects.create(user=user)
+
+        if date:
+            try:
+                created_at = datetime.strptime(date, "%Y-%m-%d %H:%M")
+                order.created_at = created_at
+            except ValueError:
+                raise ValidationError(
+                    "Date must be in the format 'YYYY-MM-DD HH:MM'."

```

Comment:
Ensure that the date format is validated correctly. The current implementation attempts to parse the date string and raises a `ValidationError` if it fails, which is a good practice.